### PR TITLE
Add support for Response trailers for netstandard20

### PR DESCRIPTION
**Pull request by simonferquel**
When using the netstandard20 version, response trailers were previously ignored. This was causing issues with handling responses from a GRPC service.

Grpc-dotnet works with WinHttpHandler on recent Windows versions, and trailers are stored in the request generic property bag as shown here: https://github.com/grpc/grpc-dotnet/blob/456098bcfeb9c796c19ea023b3ab3703181f1089/src/Shared/TrailingHeadersHelpers.cs#L30

This PR uses the same way to store trailers when compiled for netstandard 2.0

Signed-off-by: Simon Ferquel <simon.ferquel@unity3d.com>

### CODE REVIEW

**Comment by TalAloni** [comment]
Thanks, can you explain what the use-case is?
IIUC - this change is only useful for HTTP 2.0, and given that .NET 4.7.2 does not support ALPN, it is only useful for unencrypted HTTP 2.0 communication, which is not very common.
If one requires HTTP 2.0 and ALPN, this change is probably not going to help, right?

**Comment by simonferquel** [comment]
This is a Unity use case, where we have some programs running grpc services on a named pipe - without encryption (on Windows platform). Those services can be consumed in MSBuild tasks, which run on .NET 4.7.2 within Visual Studio.
This change + the connect callback allows to connect to those services :) - otherwise grpc client complains that the response is missing some trailers.

**Comment by TalAloni** [comment]
that responseMessage.RequestMessage.Properties["__ResponseTrailers"] logic seems very use-case specific... isn't it?
not sure that's the right place, and besides, the 3.1 branch is experimental anyway

**Comment by tgjones** [comment]
> that responseMessage.RequestMessage.Properties["__ResponseTrailers"] logic seems very use-case specific... isn't it?

@TalAloni it is the "official" place to put response trailing headers when running on .NET Framework: [reference](https://github.com/dotnet/dotnet-api-docs/issues/5404#issue-827553293)

> On .NET Framework, trailers are added to `HttpRequestMessage.Properties["__ResponseTrailers"]`. Use with caution and only in well justified compatibility scenarios, since the property is obsolete

Any chance you could re-review this PR please, with that added context?

File: StandardSocketsHttpHandler/Net/Http/SocketsHttpHandler/Http2Stream.cs [hunk 1]
```diff
@@ -960,19 +960,31 @@ public async ValueTask<int> ReadDataAsync(Memory<byte> buffer, HttpResponseMessa
...
+                    responseMessage.RequestMessage.Properties["__ResponseTrailers"] = headers;
 #endif
-            }
+                    _trailers.Clear();
+
+                }
 
-            private async Task SendDataAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken)
+            }
+            class HttpTrailers : HttpHeaders
```
Comment (TalAloni) [inline StandardSocketsHttpHandler/Net/Http/SocketsHttpHandler/Http2Stream.cs]: Should be a separate file and should only be defined for NETSTANDARD2_0

File: StandardSocketsHttpHandler/Net/Http/SocketsHttpHandler/Http2Stream.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (TalAloni) [inline StandardSocketsHttpHandler/Net/Http/SocketsHttpHandler/Http2Stream.cs]: You have removed the BOM from the beginning of the file, please revert this change

File: StandardSocketsHttpHandler/Net/Http/SocketsHttpHandler/Http2Stream.cs (hunk 1 above)
Comment (tgjones) [reply StandardSocketsHttpHandler/Net/Http/SocketsHttpHandler/Http2Stream.cs]: Done 👍

File: StandardSocketsHttpHandler/Net/Http/SocketsHttpHandler/Http2Stream.cs (hunk 2 above)
Comment (tgjones) [reply StandardSocketsHttpHandler/Net/Http/SocketsHttpHandler/Http2Stream.cs]: Done 👍

**Comment by tgjones** [comment]
@TalAloni thank you for merging! Please could you release a new version of the hidden 3.1.0.x package on NuGet?

**Comment by TalAloni** [comment]
Done (v3.1.0.5)

**Comment by tgjones** [comment]
Thank you!
